src/main/java/uk/ac/ebi/tsc/portal/api/team/controller/TeamRestController.java

```diff
@@ -128,20 +128,16 @@ public TeamRestController(TeamService teamService,
 	public Resources<TeamResource> getAllTeamsForCurrentUser(Principal principal){
 
 		Collection<Team> teams = teamService.findByAccountUsername(principal.getName());
-		return new Resources<>(teams.stream().map(
-				TeamResource::new
-				).collect(Collectors.toList())
-				);
+		Collection<TeamResource> teamResources = teamService.populateTeamContactEmails(teams, principal);
+		return new Resources<>(teamResources);
 	}
 
 	@RequestMapping(value="/all",method=RequestMethod.GET)
 	public Resources<TeamResource> getAllTeams(Principal principal){
 
 		Collection<Team> teams = teamService.findAll();
-		return new Resources<>(teams.stream().map(
-				TeamResource::new
-				).collect(Collectors.toList())
-				);
+		Collection<TeamResource> teamResources = teamService.populateTeamContactEmails(teams, principal);
```

Comment:
Performance killer - will you be getting team emails one by one for each team in the system? Do we need team emails here at all?

Reply:
not needed here too.

Reply:
This is done